tests/baselines/reference/privateNameStaticFieldUnaryMutation.js

```diff
@@ -0,0 +1,82 @@
+//// [privateNameStaticFieldUnaryMutation.ts]
+class C {
+    static #test: number = 24;
+    constructor() {
+        C.#test++;
+        C.#test--;
+        ++C.#test;
+        --C.#test;
+        const a = C.#test++;
+        const b = C.#test--;
+        const c = ++C.#test;
+        const d = --C.#test;
+        for (C.#test = 0; C.#test < 10; ++C.#test) {}
+        for (C.#test = 0; C.#test < 10; C.#test++) {}
+    }
+    test() {
+        this.getClass().#test++;
+        this.getClass().#test--;
+        ++this.getClass().#test;
+        --this.getClass().#test;
+        const a = this.getClass().#test++;
+        const b = this.getClass().#test--;
+        const c = ++this.getClass().#test;
+        const d = --this.getClass().#test;
+        for (this.getClass().#test = 0; this.getClass().#test < 10; ++this.getClass().#test) {}
+        for (this.getClass().#test = 0; this.getClass().#test < 10; this.getClass().#test++) {}
+    }
+    getClass() { return C; }
+}
+
+
+//// [privateNameStaticFieldUnaryMutation.js]
+var __classStaticPrivateFieldGet = (this && this.__classStaticPrivateFieldGet) || function (receiver, classConstructor, propertyDescriptor) {
+    if (receiver !== classConstructor) {
+        throw new TypeError("Private static access of wrong provenance");
+    }
+    if (propertyDescriptor === undefined) {
+        throw new TypeError("Private static field was accessed before its declaration.");
+    }
+    return propertyDescriptor.value;
+};
+var __classStaticPrivateFieldSet = (this && this.__classStaticPrivateFieldSet) || function (receiver, classConstructor, propertyDescriptor, value) {
+    if (receiver !== classConstructor) {
+        throw new TypeError("Private static access of wrong provenance");
+    }
+    if (propertyDescriptor === undefined) {
+        throw new TypeError("Private static field was accessed before its declaration.");
+    }
+    propertyDescriptor.value = value;
+    return value;
+};
+var _C_test;
+class C {
+    constructor() {
+        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
+        __classStaticPrivateFieldSet(_a = C, C, _C_test, +__classStaticPrivateFieldGet(_a, C, _C_test) + 1);
+        __classStaticPrivateFieldSet(_b = C, C, _C_test, +__classStaticPrivateFieldGet(_b, C, _C_test) - 1);
+        __classStaticPrivateFieldSet(_c = C, C, _C_test, +__classStaticPrivateFieldGet(_c, C, _C_test) + 1);

```

Comment:
This is not specific to private statics but it's a bit weird that `C.#test++` and `++C.#test` evaluates to the same thing as standalone expressions. They are correctly handled if the expression is used, though.